### ISSUE TITLE
Add Plots of Raw and Resampled Data

### ISSUE THREAD

**Issue by jacobbieker** [openclimatefix/pv-solar-farm-forecasting#1]
Make plots of the raw data, resampled data, and other individual PV sites to see if the solar farms act the same as our other PV datasets

## Detailed Description
This could go into a one-pager to give a quick overview of the dataset

## Possible Implementation
Just a few simple plots of 

- [ ] Raw and resampled values for a few days through the year
- [ ] Average raw and resampled values over the whole dataset for each time of day (i.e. the general shape of the raw data)
- [ ] Update frequency, and if it changes through the year, as in is the site having more delta changes in summer vs winter
- [ ] How many bad data entries there are, and how often they show up

**Comment by peterdudfield** [comment]
@jacobbieker this looks super good and clear. 

Bonus: the one page could be a quick Markdown document with the plots saved as images and with a bit of text around them

**Comment by vrym2** [comment]
From ECR [register](https://www.ukpowernetworks.co.uk/electricity/distribution-energy-resources/the-embedded-capacity-register) provided by @jacobbieker , a meta data file can be downloaded which contains nearly 1000 post codes of different pv sites and their attributes, and I assume the pv outputs of these sites can be extracted from pvoutput.org? @jacobbieker @peterdudfield

**Comment by jacobbieker** [comment]
No, the only output is the file I shared with you, these sites wont be on pvoutput.org, its just data from that provider

**Comment by vrym2** [comment]
`ukpowernewtorks` provides an API for all their datasets, including for the one @jacobbieker has sent me the link to `xlsx` file, here is the [link](https://ukpowernetworks.opendatasoft.com/explore/dataset/embedded-capacity-register/api/?disjunctive.licence_area&refine.grid_supply_point=CANTERBURY+NORTH&refine.energy_conversion_technology_1=Photovoltaic) to the dataset we need for the site @jacobbieker sent me

Using the `api` link, i have written a simple script to check the status of `api_repsone` and to get the first record of the dataset.

**Comment by vrym2** [comment]
![image](https://user-images.githubusercontent.com/93340339/211590776-1e41c4a4-b26b-4b7a-99b4-d8182b0b6f97.png)

![image](https://user-images.githubusercontent.com/93340339/211590957-68cf6334-0671-47b6-a3d5-ec6d1c6bb2b4.png)

@jacobbieker Plotting of raw data and resampled  5-min interval data of first 100 entries, the resampling is done by considering mean of the 5-minute intervals. 

I believe that has not lost as there is little difference between both plots

**Comment by jacobbieker** [comment]
Good job! Is there a chance you can plot the errors? So plot the top one divided by the bottom one?

**Comment by vrym2** [comment]
> Good job! Is there a chance you can plot the errors? So plot the top one divided by the bottom one?

Yes, we can do that, and as per your suggestion we can do that for each day in the entire time series, but bear in mind that even after resampling, the 5 min intervals are not contiguous!

**Comment by jacobbieker** [comment]
I think it means only those 26 timesteps are bad, not that the data values are bad. This resampling should make it continuous overall.

**Comment by peterdudfield** [comment]
Is `bad-data` something that is negative. Could you just remove that data? Would be good to get a count of them, ie. what % of them. Might want to remove a few hours around the bad data to be sure.

**Comment by vrym2** [comment]
@peterdudfield @jacobbieker 
That `Bad Data` is one of the two columns in the `csv file` we got from Dan by email, `ocf-email` - [link](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), the entire column consists of only negative values and `np.zeros`, but these `negative values` are repetitive, meaning they are discrete/categorical variables, not a random values, it is possible to interpolate them in the missing time steps, but not by conventional statistical methods for example with `mean` or `median` etc.

**Comment by jacobbieker** [comment]
Okay, if that's the case, then I would go with drop the data first with the bad data label, and then interpolate between the remaining ones to a 5 minutely timeseries, not with mean or median, but just an interpolation as a first pass

**Comment by vrym2** [comment]
After resampling, the plots between original raw data and resampled data, the resampling of the 5-minutey intervals is done by grouping the timesteps falling under each 5 minute interval (eg: 11/01/2023 15:31:11 and 11/01/2023 15:34:23) and taking their minimum value.

![image](https://user-images.githubusercontent.com/93340339/211847297-f958fbef-7ddb-420f-82cd-b52bbd5d3df1.png)

![image](https://user-images.githubusercontent.com/93340339/211847345-9cc202a3-d0d6-4265-a5d9-b816f1352cc3.png)

**Comment by jacobbieker** [comment]
Okay, could you do it without taking the minimum, just doing an interpolation of the values to 5 minutely values?

**Comment by vrym2** [comment]
> Okay, could you do it without taking the minimum, just doing an interpolation of the values to 5 minutely values?

Interpolation into 5 minutely intervals has been done! Please check the PR

**Comment by jacobbieker** [comment]
Okay, I've left some more comments on there. If we choose to not resample, then in general the raw data shouldn't be touched, other than to remove the bad data values, as we want to keep the raw data as close to what is coming out of the solar farm as possible. And with resampling, the comments are in the PR, but generally, shouldn't round values to the resample period we want, as that will mess up the interpolation.

**Comment by vrym2** [comment]
> Okay, I've left some more comments on there. If we choose to not resample, then in general the raw data shouldn't be touched, other than to remove the bad data values, as we want to keep the raw data as close to what is coming out of the solar farm as possible. And with resampling, the comments are in the PR, but generally, shouldn't round values to the resample period we want, as that will mess up the interpolation.

Pleas check the PR. I have written an interpolation function and tested it.

**Comment by jacobbieker** [comment]
Cool, nice job, could you share the output plots?

**Comment by vrym2** [comment]
> Cool, nice job, could you share the output plots?

These are the plots from randomly selected date

![image](https://user-images.githubusercontent.com/93340339/212112572-df3d8e4b-9e99-48b4-968c-b04d9225e1e1.png)

![image](https://user-images.githubusercontent.com/93340339/212112623-b6105055-acf1-4b38-9894-4df622288b25.png)

**Comment by peterdudfield** [comment]
Whats the yaxis on these? 'bad data' seems a odd label?

**Comment by vrym2** [comment]
It is the negative data values we got from the client in this [email](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), in that there are some entries as `Bad Data` alongside negative values, those negative values supposed to be representing delta change if I'm right!

**Comment by peterdudfield** [comment]
> It is the negative data values we got from the client in this [email](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), in that there are some entries as `Bad Data` alongside negative values, those negative values supposed to be representing delta change if I'm right!

It would be great to get a yaxis with `kw` on it? Do you think that is possible? you might have to add those delta values up to make kw.

**Comment by peterdudfield** [comment]
I would have thought Bad_data is a flag of 0 or 1, depending if the data is good or not. People use this to mark bad data, but not delete it

**Comment by vrym2** [comment]
The plot of cumulative sum over time for these values in a single day before and after interpolation. I asked Dan to ask them for more information on how we can interpret this data they provided.

![image](https://user-images.githubusercontent.com/93340339/212300861-4c86a5a1-b090-452e-b959-0d86cc89c6df.png)

![image](https://user-images.githubusercontent.com/93340339/212300906-1f18ff98-5649-4e16-87a4-2fbe16386a10.png)


> > It is the negative data values we got from the client in this [email](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), in that there are some entries as `Bad Data` alongside negative values, those negative values supposed to be representing delta change if I'm right!
> 
> It would be great to get a yaxis with `kw` on it? Do you think that is possible? you might have to add those delta values up to make kw.

**Comment by jacobbieker** [comment]
> > It is the negative data values we got from the client in this [email](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), in that there are some entries as `Bad Data` alongside negative values, those negative values supposed to be representing delta change if I'm right!
> 
> It would be great to get a yaxis with `kw` on it? Do you think that is possible? you might have to add those delta values up to make kw.

It does seem that the values are only 0 or negative. We might need to ask how this delta becomes `kw` as I would have assumed there would need to be some positive values in the data, which there isn't. Unless the 0 values are the max capacity or something, and its going down from there?

**Comment by peterdudfield** [comment]
> > > It is the negative data values we got from the client in this [email](https://mail.google.com/mail/u/1/#inbox/FMfcgzGrbRTNSdptbtKPDctdSgMKgsNr), in that there are some entries as `Bad Data` alongside negative values, those negative values supposed to be representing delta change if I'm right!
> > 
> > 
> > It would be great to get a yaxis with `kw` on it? Do you think that is possible? you might have to add those delta values up to make kw.
> 
> It does seem that the values are only 0 or negative. We might need to ask how this delta becomes `kw` as I would have assumed there would need to be some positive values in the data, which there isn't. Unless the 0 values are the max capacity or something, and its going down from there?

Yea, perhaps we need to get on a call with them, something about this data is all over the place

**Comment by peterdudfield** [comment]
Its not just some weird days is it? Are there normal days? 

Do we have any site metdata to help us here? Like whats the capacity of the site?

**Comment by jacobbieker** [comment]
The capacity is  39MW or 41.053MVA, and we have the location, and there isn't any normal days, in terms of, there are no non-negative values in the CSV we got.
